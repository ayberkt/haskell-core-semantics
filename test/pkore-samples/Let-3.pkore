let(nonRec(tmVar([type omitted], name(foo)),
           app(lam(tmVar([type omitted], name(a)),
                   lam(tmVar([type omitted], name(b)),
                       app(var(tmVar([type omitted], name(a))), var(tmVar([type omitted], name(b)))))),
               lam(tmVar([type omitted], name(x)), var(tmVar([type omitted], name(x)))))),
    app(var(tmVar([type omitted], name(foo))), lit(litInt(3, [type omitted]))))
