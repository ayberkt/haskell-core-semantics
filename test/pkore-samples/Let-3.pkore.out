<generatedTop> <k> val ( 5 ) </k> <store> 1 |-> indirection ( 4 ) 2 |-> closure ( name ( a ) , lam ( tmVar ( [type omitted] , name ( b ) ) , app ( var ( tmVar ( [type omitted] , name ( a ) ) ) , var ( tmVar ( [type omitted] , name ( b ) ) ) ) ) , .Map ) 3 |-> indirection ( 7 ) 4 |-> closure ( name ( b ) , app ( var ( tmVar ( [type omitted] , name ( a ) ) ) , var ( tmVar ( [type omitted] , name ( b ) ) ) ) , name ( a ) |-> 3 ) 5 |-> litObj ( lit ( litInt ( 17 , [type omitted] ) ) ) 7 |-> closure ( name ( x ) , var ( tmVar ( [type omitted] , name ( x ) ) ) , .Map ) </store> <env> .Map </env> <exit-code> 0 </exit-code> </generatedTop>
